lib/routes/pconline/focus.ts

```diff
@@ -0,0 +1,143 @@
+import { Route } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+const rootUrl = 'https://www.pconline.com.cn';
+interface Item {
+    id: string;
+    title: string;
+    channelName: string;
+    wap_3g_url: string;
+    bigImage: string;
+    cover: string;
+    largeImage: string;
+    authorname: string;
+    authorImg: string;
+    pc_pubDate: string;
+    artType: string;
+    summary: string;
+    url: string;
+}
+
+const categories = {
+    all: {
+        title: '全部',
+        path: '',
+    },
+    tech: {
+        title: '科技',
+        path: 'tech/',
+    },
+    finance: {
+        title: '财经',
+        path: 'finance/',
+    },
+    life: {
+        title: '生活',
+        path: 'life/',
+    },
+    company: {
+        title: '公司',
+        path: 'company/',
+    },
+    character: {
+        title: '人物',
+        path: 'character/',
+    },
+};
+
+const getContent = (link: string) => cache.tryGet(link, async () => {
+        const detailResponse = await got({
+            method: 'get',
+            url: `https:${link}`,
+            responseType: 'arrayBuffer',
+        });
+
+        const utf8decoder = new TextDecoder('GBK');
+        const html = utf8decoder.decode(detailResponse.data);
+        const $ = load(html);
+        const content = $('.context-box .context-table tbody td').html() || '';
+        return content;
+    });
+
+export const handler = async (ctx) => {
+    const { category = 'all' } = ctx.req.param();
+    const cate = categories[category] || categories.all;
+    const currentUrl = `${rootUrl}/3g/other/focus/${cate.path}index.html`;
+
+    const response = await got({
+        method: 'get',
+        url: currentUrl,
+    });
+
+    const resString = response.data
+        .replace(/Module\.callback\((.*)\)/s, '$1')
+        .split('\n')
+        .filter((e) => e.indexOf('"tags":') !== 0)
+        .join('\n')
+        .replaceAll("'", '"');
+    const tinyData = resString.replaceAll(/[\n\r]/g, '');
+    const dataString = tinyData.replaceAll(',}', '}');
+    const data = JSON.parse(dataString || '');
+    const { articleList } = data;
+    const items = articleList.map(async (item: Item) => {
+        const content = await getContent(item.url);
+        return {
+            id: item.id,
+            title: item.title,
+            author: [
+                {
+                    name: item.authorname,
+                    avatar: item.authorImg,
+                },
+            ],
+            pubDate: parseDate(item.pc_pubDate),
+            link: item.url,
+            description: content || item.summary,
+            category: item.channelName,
+            image: item.cover,
+        };
```

Comment:
Please cache the whole returned object instead of `content` only

Reply:
sorry, I don't understand the return object, `content` is the entire article HTML

Reply:
> whole returned object
```js
return {
    id: ...,
    title: ...,
    ...
}
```

Reply:
Okay, thank you.